Src/FluentAssertions/Common/TaskTimer.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace FluentAssertions.Common
+{
+    /// <summary>
+    /// Default implementation for <see cref="ITimer"/> for production use.
+    /// </summary>
+    internal class TaskTimer : ITimer
+    {
+        public Task DelayAsync(TimeSpan delay, CancellationToken cancellationToken)
+        {
+            return Task.Delay(delay, cancellationToken);
+        }
+
+        public bool Wait(Task task, TimeSpan timeout)
+        {
+            using (NoSynchronizationContextScope.Enter())
```

Comment:
Is there any problem in Tasks being nested twice in a `NoSynchronizationContextScope`.
First in `Should` and then again here?

Reply:
It's fine if you nest them. It'll just get the `null` context, set it to `null` and then restore the `null` ;-)